src/Core/OrganizationFeatures/Groups/DeleteGroupCommand.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+
+namespace Bit.Core.OrganizationFeatures.Groups;
+
+public class DeleteGroupCommand : IDeleteGroupCommand
+{
+    private readonly IGroupRepository _groupRepository;
+    private readonly IEventService _eventService;
+
+    public DeleteGroupCommand(IGroupRepository groupRepository, IEventService eventService)
+    {
+        _groupRepository = groupRepository;
+        _eventService = eventService;
+    }
+
+    public async Task DeleteAsync(Group group)
+    {
+        await _groupRepository.DeleteAsync(group);
+        await _eventService.LogGroupEventAsync(group, Enums.EventType.Group_Deleted);
+    }
+
+    public async Task DeleteManyAsync(IEnumerable<Group> groups)
+    {
+        var groupsToDelete = groups as Group[] ?? groups.ToArray();
```

Comment:
I'm still a bit hung up on this. I think I understand the general concern: if `groups` is an `IEnumerable` that defers enumeration and does not cache enumeration results, then this will read from the database each time we enumerate it in the method (i.e. in the 2 `Select` statements below).

At the moment, we generally pass around `IEnumerables` without regard for this risk, so is this a blind spot in our practices? Should we just force enumeration in repository methods to be safe? I don't know, I think this is worth raising with the team.

For now, in this PR, I think this should be enforced by method parameter types - it's basically a type system concern. If the method requires a list/array/whatever then it should state that in its method signature.

Our repository methods return `ICollection`, is this subject to deferred execution? I'm trying to think of the most convenient/general parameter type to require. What do you think?

Reply:
> At the moment, we generally pass around `IEnumerables` without regard for this risk, so is this a blind spot in our practices?

I believe this might be a blind spot, however, I don't believe we typically use deferred execution, so it's likely not a significant concern. Should we start using deferred execution more frequently, I think it would be beneficial to consider / keep in mind going forward.

> Our repository methods return `ICollection`, is this subject to deferred execution? I'm trying to think of the most convenient/general parameter type to require. What do you think?

Yep, `ICollection` makes the most sense to me as well, and is not subject to deferred execution. I went ahead and switched the `DeleteManyAsync` signature to use `ICollection<Group>` and removed the `.toArray()`  in the method body.